### PR TITLE
SISRP-7615 Create basic DEMO for QA Test Automation framework

### PR DESCRIPTION
**Pull request by mDibyo**
https://jira.berkeley.edu/browse/SISRP-7615

### CODE REVIEW

**Comment by christianvuerings** [comment]
@ctweney could you also have a look at this PR?

**Comment by mDibyo** [comment]
I will add another commit updating the README with instructions for creating a config file with credentials.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,112 @@
+### JetBrains template
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio
+
+*.iml
+
+## Directory-based project format:
+.idea/
+# if you remove the above rule, at least ignore the following:
+
+# User-specific stuff:
```
Comment (christianvuerings) [inline .gitignore]: @mDibyo could we remove the commented out lines that don't provide any value / info?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+# Sphinx documentation
+docs/_build/
+
+# PyBuilder
+target/
+
+# Created by .ignore support plugin (hsz.mobi)
+
+# Configuration files
+test/.config.json
```
Comment (christianvuerings) [inline .gitignore:89]: @mDibyo could you provide an example? e.g. `.example.config.json`. This way we'll know the format of the `.config.json file`

File: docker/chrome-browser/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        libxss1 \
+        libappindicator1 \
+        libindicator7 \
+        wget
+RUN wget -O opt/google-chrome-stable_current_amd64.deb \
+        https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+RUN dpkg -i opt/google-chrome*.deb
+RUN apt-get install -y -f
+RUN dpkg -i opt/google-chrome*.deb
+
```
Comment (christianvuerings) [inline docker/chrome-browser/Dockerfile]: @mDibyo can we remove the extra line breaks? (There should only be one at the end of a file)

File: test/pytest/test_browser.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    driver.get(config["website_url"])
+
+    email_element = driver.find_element_by_id('userid')
+    email_element.send_keys(config["userid"])
+
+    password_element = driver.find_element_by_id('pwd')
+    password_element.send_keys(config["password"])
+    password_element.submit()
+
+    WebDriverWait(driver, 10).until(EC.title_contains(config["title"]))
```
Comment (christianvuerings) [inline test/pytest/test_browser.py]: @mDibyo looks like GitHub is complaining about an extra line break at the end, could we add that?

File: README.md [hunk 5]
```diff
@@ -1 +1,20 @@
...
+4. Add file test/.config.json with the following information:
+```json
+{
+    "website_url": "<Campus Solutions Test Instance URL>",
+    "userid": "<User ID for Test Instance>",
+    "password": "<Password for Test Instance>",
+    "title": "Berkeley Student Information System"
+}
+```
+3. Run `run.sh` (may required sudo. If using external test directory, run as `sudo -E ./run.sh` to
```
Comment (christianvuerings) [inline README.md]: @mDibyo the numbering seems to be a bit messed up, could you check whether it works if we just use `1.` everywhere (even for the 2nd / 3th / ... items)?
Also, you might have to use 4 spaces for indenting the JSON - https://github.com/ets-berkeley-edu/calcentral/edit/master/README.md has some really good examples for this.

File: docker/firefox-browser-gui/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,17 @@
+FROM ubuntu:14.04
```
Comment (christianvuerings) [inline docker/firefox-browser-gui/Dockerfile:1]: @mDibyo as discussed, it would be great if we could avoid some of the duplication in the Dockerfiles

**Comment by ctweney** [comment]
This looks great! Nice work!

File: test/pytest/test_browser.py (hunk 4 above)
Comment (mDibyo) [inline test/pytest/test_browser.py]: Done

File: docker/firefox-browser-gui/Dockerfile (hunk 6 above)
Comment (mDibyo) [inline docker/firefox-browser-gui/Dockerfile:1]: Done.

File: .gitignore (hunk 1 above)
Comment (mDibyo) [inline .gitignore]: Done

File: .gitignore (hunk 2 above)
Comment (mDibyo) [inline .gitignore:89]: As we discussed, added a note in README.

File: docker/chrome-browser/Dockerfile (hunk 3 above)
Comment (mDibyo) [inline docker/chrome-browser/Dockerfile]: Done.

**Comment by mDibyo** [comment]
I have responded to all the comments.

I have also set up the ucberkeley account on Docker Hub so that whenever a commit is pushed to master on Github, the images ucberkeley/firefox-browser-gui and ucberkeley/sis-qa-test-auto are automatically rebuilt from the repository.

**Comment by christianvuerings** [comment]
lgtm @ctweney want to do a last quick check?

**Comment by ctweney** [comment]
Pull LGTM. 

I'm concerned about the ucberkeley Docker Hub account. Who has the login credentials for that? Something called "ucberkeley" should be owned by an administrative entity, not a single person. We might need to get ETS Operations to adopt the Docker Hub account management.

**Comment by christianvuerings** [comment]
@ctweney afaik there's actually no password for the 'ucberkeley' group, it's just the owners that have access to it and can make changes, very similar to organizations on GitHub.

**Comment by mDibyo** [comment]
Christian and me created an 'organization' for UC Berkeley on Docker Hub. This is not an account, rather, it provides access to the Docker repositories stored under the ucberkeley tag by defining a set of groups. Each group can have three levels of access: Read, Write, Admin. Anyone added to a group with Admin access can make changes to the organization on Docker Hub.
Currently, there are two groups, 'owners' which is the default group and has Admin access, and 'sis', which I defined, and has write access to the Docker repositories. Both of them currently have Christian and me as members.
Going with your suggestion, it might be best to add someone from ETS Operations to the 'owners' group. Then, remove other members from 'owners' as necessary (only Admins can create groups and add/remove members to/from groups). Since, we might be adding/removing more people to 'sis' (and other groups we may define), at least one of us should still remain on the 'owners' group.

**Pull request merged**